lib/OpenQA/Scheduler/Scheduler.pm

```diff
@@ -793,8 +793,28 @@ sub job_cancel($;$) {
         $new_result = OpenQA::Schema::Result::Jobs::OBSOLETED;
         # 'monkey patch' cond to be useable in chained search
         $cond{'me.id'} = delete $cond{id};
-        # filter out all that have any comment (they are considered 'important')
+        # filter out all jobs that have any comment (they are considered 'important') ...
         $jobs_to_cancel = $scheduled_jobs->search({'comments.job_id' => undef}, {join => 'comments'});
+        # ... or belong to a tagged build, i.e. is considered important
+        # this might be even the tag 'not important' but not much is lost if
+        # we still not cancel these builds
+        my $groups_query = $scheduled_jobs->get_column('group_id')->as_query;
+        my $groups       = schema->resultset("JobGroups")->search($groups_query);
+        my @important_builds;
+        while (my $g = $groups->next) {
+            for my $comment ($g->comments->all) {
+                my @tag = $comment->tag;
+                push @important_builds, $tag[0];
```

Comment:
It's very early and I only had s small coffee yet, but to me these 7 lines look to me like doing the same as these 2:
```
my $comments = schema->resultset("Comments")->search({group_id => -in => $groups_query});
my @important_builds = map { $_->tag } $comments->all;
```